### PR TITLE
tagging intents by intent name

### PR DESCRIPTION
**Pull request by faya98**

### CODE REVIEW

File: tagging_intents_by_intent_name [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+              help='HumanFirst playbook id')
+def main(delimiter: str, add_tag: str,
+         delete_tag: str, search_string: str,
+         username: str, password: str,
+         namespace: str, playbook: str):
+    '''Main function'''
+
+    if add_tag == "" and delete_tag == "":
+        error_msg = "Both add_tag and delete_tag are empty. Atleast one should be provided"
+        raise  Exception(error_msg) # pylint: disable=broad-exception-raised
```
Comment (sbroadhurst-hf) [inline tagging_intents_by_intent_name:38]: Something I found out recently.  If you raise your custom exceptions as RunTimeErrors then you don't need to declare them explicitly and will pass this check
https://docs.python.org/3/library/exceptions.html

```
exception RuntimeError
Raised when an error is detected that doesn’t fall in any of the other categories. The associated value is a string indicating what precisely went wrong
```

File: tagging_intents_by_intent_name [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+            msg1 = f"\'{add_hf_tag.name}\' is created for the playbook - \'{playbook}\'"
+            msg2 = f"in the namespace - \'{namespace}\'"
+            print(f"{msg1} {msg2}")
+    else:
+        add_hf_tag = labelled_workspace.tag(tag=add_tag)
+
+    # remove color from add_hf_tag
+    # this is beacuse the tags inside intent should have only the id and name
+    add_hf_tag = add_hf_tag.to_dict()
```
Comment (sbroadhurst-hf) [inline tagging_intents_by_intent_name:65]: This seems like a workaround around something in the way the object should appear which should be better handled by humanfirst module.

File: tagging_intents_by_intent_name [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        del_flag = 0
+        if intent_name.find(search_string) != -1:
+            # check if the tag is already present in the intent
+            intent_tags = labelled_workspace.intents[intent_name].tags
+            intent_tags_after_del = []
+            for _, intent_tag in enumerate(intent_tags):
+                intent_tag = intent_tag.to_dict()
+
+                # color peroperty should be removed from the tags inside intent property
+                # TODO: update humanfirst objects to not set the color property in the tags inside intent property
```
Comment (sbroadhurst-hf) [inline tagging_intents_by_intent_name:79]: Yep this TODO is the right thing

File: tagging_intents_by_intent_name [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+                intent_tags_after_del.append(intent_tag)
+
+            labelled_workspace.intents[intent_name].tags = intent_tags_after_del
+
+            if del_flag == 0 and delete_tag != "":
+                error_msg = f"{delete_tag} is not present in any of the intents"
+                raise Exception(error_msg) # pylint: disable=broad-exception-raised
+
+            if add_flag == 0 or del_flag == 1:
```
Comment (sbroadhurst-hf) [inline tagging_intents_by_intent_name:101]: recommend elif, then an elif on del_flag = 0 and add_flag = 0 which is the nothing to do correct step, then catch all else which throws an error if it isn't caught by the above ones catches any logical errors and throws and exception.

File: tagging_intents_by_intent_name [hunk 5]
```diff
@@ -0,0 +1,117 @@
+"""
+tagging intents by using intent name as the search string
+
+Set HF_USERNAME and HF_PASSWORD as environment variables
+"""
```
Comment (sbroadhurst-hf) [inline tagging_intents_by_intent_name:5]: Needs an explanation of what it does and how

File: tagging_intents_by_intent_name (hunk 3 above)
Comment (faya98) [reply tagging_intents_by_intent_name:86]: Raised a ticket in CS https://zia-ai.atlassian.net/browse/CS-32

**Review by sbroadhurst-hf — APPROVED** [review]
TQ for updates.

**Pull request merged**